android/src/toga_android/dialogs.py

```diff
@@ -130,25 +132,39 @@ def __init__(
 
 
 class OpenFileDialog(BaseDialog):
+    class HandlerOpenDialog(uf.HandlerFileDialog):
+        def show(self):
+            intent = Intent(Intent.ACTION_OPEN_DOCUMENT)
+            intent.addCategory(Intent.CATEGORY_OPENABLE)
+            intent.setType("*/*")
+            self.app.start_activity(intent, on_complete=self.parent.handler)
+
     def __init__(
-        self,
-        title,
-        initial_directory,
-        file_types,
-        multiple_select,
+            self,
+            title,
+            initial_directory,
+            file_types,
+            multiple_select,
     ):
-        super().__init__()
+        self.native = OpenFileDialog.HandlerOpenDialog(self, toga.App.app.current_window)
+        self.mActive = self.native.mActive
 
-        toga.App.app.factory.not_implemented("dialogs.OpenFileDialog()")
-        self.native = None
+    def handler(self, code, indent):
+        self._handler(indent.getData())
+
+    def _handler(self, uri):
+        ist = self.mActive.getContentResolver().openInputStream(uri)
+        isr = uf.InputStreamReader(uf.Objects.requireNonNull(ist))
```

Comment:
InputStreamReader is a java class; it should be imported *from* the Java namespace, not implicitly from the `utilfile` namespace.

Reply:
I will change the import